network/src/node.rs

```diff
@@ -114,20 +116,29 @@ impl<S: Storage + core::marker::Sync + Send + 'static> Node<S> {
 impl<S: Storage + Send + core::marker::Sync + 'static> Node<S> {
     /// Creates a new instance of `Node`.
     pub async fn new(config: Config) -> Result<Self, NetworkError> {
-        Ok(Self(Arc::new(InnerNode {
+        let node = Self(Arc::new(InnerNode {
             id: thread_rng().gen(),
             state: Default::default(),
             local_address: Default::default(),
             config,
             inbound: Default::default(),
             peer_book: PeerBook::spawn(),
             sync: Default::default(),
+            network_topology: Default::default(),
             launched: Utc::now(),
             tasks: Default::default(),
             threads: Default::default(),
             shutting_down: Default::default(),
             master_dispatch: RwLock::new(None),
-        })))
+        }));
+
+        if node.config.is_bootnode() {
+            node.network_topology
+                .set(NetworkTopology::default())
+                .expect("network topology was set more than once");
```

Comment:
since this is in `Node::new`, we might want to use `unwrap()` with a comment or mention that it's impossible in the `expect` message